src/main/java/com/minecolonies/commands/AbstractSingleCommand.java

```diff
@@ -0,0 +1,35 @@
+package com.minecolonies.commands;
+
+import net.minecraft.command.ICommandSender;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A command that has children.
+ */
+public abstract class AbstractSingleCommand implements ISubCommand
+{
+
+    private final String[] parents;
+
+    /**
+     * Initialize this SubCommand with it's parents.
+     *
+     * @param parents an array of all the parents.
+     */
+    public AbstractSingleCommand(@NotNull String... parents)
+    {
+        this.parents = parents;
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Store a copy of "parents". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2384)